includes/Core/User_Input/REST_User_Input_Controller.php

```diff
@@ -94,20 +94,28 @@ function ( $paths ) {
 	 * @return array List of REST_Route objects.
 	 */
 	private function get_rest_routes() {
-		$can_authenticate = function() {
-			return current_user_can( Permissions::AUTHENTICATE );
-		};
-
 		return array(
 			new REST_Route(
 				'core/user/data/user-input-settings',
 				array(
 					array(
 						'methods'             => WP_REST_Server::READABLE,
 						'callback'            => function() {
-							return rest_ensure_response( $this->user_input->get_answers() );
+							$response = rest_ensure_response( $this->user_input->get_answers() );
+
+							if ( ! current_user_can( 'list_users' ) ) {
+								foreach ( $response->data as &$setting ) {
+									if ( isset( $setting['author'] ) ) {
+										unset( $setting['author'] );
+									}
+								}
```

Comment:
It might be worth adding a comment as to _why_ we're only deleting the local variable here (because it's being passed by reference). It isn't immediately obvious from just the code.

Reply:
Added a code comment ✅